__tests__/app.test.js

```diff
@@ -86,15 +86,33 @@ describe("/api/articles", () => {
         });
       });
   });
-  test("GET 404: responds with an error when wrong endpoint is requested", () => {
+  test("GET 200: responds with an array of all articles filtered by topic if topic query value is specified", () => {
     return request(app)
-      .get("/api/banana")
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((response) => {
+        response.body.articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("GET 404: responds with an error if no articles are found for topic query value", () => {
+    return request(app)
+      .get("/api/articles?topic=7777")
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toBe("Invalid path");
+        expect(response.body.msg).toBe("No articles found");
       });
```

Comment:
Nice work on testing so far, could we also consider what would happen if we queried a valid topic that happens to have no articles associated with it? This should be a 200 but an empty response, not an error.

Reply:
Added advanced error handling for this task